Python/pylifecycle.c

```diff
@@ -1454,6 +1468,10 @@ new_interpreter(PyThreadState **tstate_p)
         return _PyStatus_ERR("Py_Initialize must be called first");
     }
 
+    if (runtime->interpreters.finalizing) {
```

Comment:
It seems like we would want this to be in effect at any point that subinterpreters should not be created.  "finalizing" implies a much more limited condition.  If we were to use a `ready`/`available` flag (as suggested above) then the semantics are more clear to readers.  Furthermore, we could drop the `runtime->initialized` check a few lines up and rely exclusively on the new flag.

Reply:
Ok, I've removed the check above and made the error message more generic too.